ostree_upload_server/bundle_importer.py

```diff
@@ -213,141 +42,19 @@ def import_bundle(bundle,
-                                                       cancellable=None)
-
-            # TODO: Handle signature requirements
-            if gpg_verify_result and gpg_verify_result.count_valid() > 0:
-                logging.info("Valid flatpak signature found")
-            else:
-                raise Exception("Bundle does not have valid signature!")
-
-            # Compare installed and header metadata, remove commit if mismatch
-            metadata_contents = BundleImporter._get_metadata_contents(repo, commit)
-            if metadata_contents == metadata['metadata']:
-                logging.debug("Committed metadata matches the static delta header")
-            else:
-                raise Exception("Committed metadata does not match the static delta header")
-
-            # Copy the commit to get correct collection and ref bindings
-            new_commit = BundleImporter._copy_commit(repo, commit,
-                                                     metadata['ref'])
-
-            # Sign the commit
-            if sign_key:
-                logging.info("Signing with key %s from %s", sign_key, gpg_homedir)
-                try:
-                    repo.sign_commit(commit_checksum=new_commit,
-                                     key_id=sign_key,
-                                     homedir=gpg_homedir,
-                                     cancellable=None)
-                except GLib.Error as err:
-                    if err.matches(Gio.io_error_quark(), Gio.IOErrorEnum.EXISTS):
-                        # Already signed with this key
-                        logging.debug("Already signed with key %s", sign_key)
-                    else:
-                        raise
-
-            # Set the ref to the new commit. Ideally this would use
-            # transaction_set_collection_ref, but that's not available
-            # on SOMA and the commit was set to use this repo's
-            # collection ID, so it wouldn't make any difference.
-            repo.transaction_set_ref(None, metadata['ref'], new_commit)
-
-            # Commit the transaction
-            repo.commit_transaction(None)
-
-        except:
-            repo.abort_transaction(None)
-            raise
-
-        # Update the repo metadata (summary, appstream, etc), but no
-        # pruning or delta generation to make it fast
-        cmd = ['flatpak', 'build-update-repo']
-        if gpg_homedir:
-            cmd.append('--gpg-homedir=' + gpg_homedir)
-        if sign_key:
-            cmd.append('--gpg-sign=' + sign_key)
-        cmd.append(repository)
-
-        logging.info('Updating repository metadata')
-        logging.debug('Executing %s', ' '.join(cmd))
-
-        subprocess.check_call(cmd)
+        for arg in inspect.getargspec(BundleImporter.import_bundle)[0]:
+            logging.info("Set %s = '%s'", arg, locals()[arg])
+
+        # Find the appropriate importer based on mimetype
+        mime_type = magic.from_file(bundle, mime=True)
```

Comment:
My python-magic has no `from_file` method. Oh, awesome. There's a different python-magic on pypi than what's in debian and fedora.

Reply:
This one is a bummer. I am on 18.04 LTS so I used pypi but I'll try to run this in a container to see if I can find compatible APIs.

Reply:
We can stick with pypi python-magic, but pypi file-magic is equivalent to the python-magic debian package. Or not. Apparently, it got removed in preference to yet the same python-magic implementation - http://metadata.ftp-master.debian.org/changelogs/main/f/file/file_5.32-2_changelog.

Now debian's python-magic corresponds to https://github.com/ahupp/python-magic/, which is https://pypi.python.org/pypi/python-magic. So, I guess the best thing to do is use python-magic from pypi with a `TODO` in `Dockerfile` and in the ansible playbook to switch to the python-magic package for 18.04.